presto-main/src/main/java/com/facebook/presto/operator/aggregation/arrayagg/ArrayAggregationFunction.java

```diff
@@ -137,7 +141,13 @@ public static void output(Type elementType, ArrayAggregationState state, BlockBu
         }
         else {
             BlockBuilder entryBuilder = out.beginBlockEntry();
-            state.forEach((block, position) -> elementType.appendTo(block, position, entryBuilder));
+            state.forEach((block, position) -> {
+                elementType.appendTo(block, position, entryBuilder);
+
+                if (entryBuilder.getRetainedSizeInBytes() > MAX_ENTRY_RETAINED_SIZE) {
+                    throw new PrestoException(CELL_TOO_LARGE, format("Array size %s exceeds the maximum supported array size %s", entryBuilder.getRetainedSizeInBytes(), MAX_ENTRY_RETAINED_SIZE));
```

Comment:
Maybe consider using `DataSize` in format string so we can use the nice toString which includes units?

Reply:
changed.. Does it then make sense to use ```DataSize``` everywhere and not just the format String?

Reply:
I would avoid using it in the hot areas of this method